dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeFactory.java

```diff
@@ -142,40 +157,85 @@ public static void deleteMultiTreeByParent1(Identifier parent, Long languageId)
                     .append("AND c.lang = ").append(languageId).append(");");
 
             db.executeStatement(query.toString());
+            
+            updateVersionTs(parent.getId(), languageId);
+            refreshPageInCache(parent.getId(), languageId);
+            
         } catch (SQLException e) {
             throw new DotDataException("Error deleting tree and multi-tree dependencies.", e);
+        } catch (DotContentletStateException e) {
+            throw new DotContentletStateException("Error deleting tree and multi-tree dependencies.", e);
+        } catch (DotSecurityException e) {
+            throw new DotSecurityException("Error deleting tree and multi-tree dependencies.", e);
         }
     }
+    
+    /**
+     * Deletes multi-tree relationships given a MultiTree object.
+     * 
+     * @param multiTree
+     * @throws DotDataException 
+     * @throws DotSecurityException 
+     *            
+     */
+    public static void deleteMultiTree (MultiTree o) throws DotDataException, DotSecurityException {
+        deleteMultiTree(o, APILocator.getLanguageAPI().getDefaultLanguage()
+                .getId());
+    }
 
+    /**
+     * Deletes multi-tree relationships given a MultiTree object and a Language Id.
+     * A language id is passed in so cleanup of cached resources of parent Page content
+     * Needs to be cleaned, along with update its version_ts value
+     * 
+     * @param multiTree
+     * @param languageId 
+     * @throws DotDataException 
+     * @throws DotSecurityException 
+     *            
+     */
+	public static void deleteMultiTree(MultiTree o, Long languageId) throws DotDataException, DotSecurityException {
+	    try {
+	        String id = o.getParent1();
+	        HibernateUtil.delete(o);
+	        updateVersionTs(id, languageId);
+	        refreshPageInCache(id,languageId);
+	        return;	        
+	    } catch (DotHibernateException e) {
+	        Logger.error(MultiTreeFactory.class,"deleteMultiTree failed:"+e,e);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "deleteMultiTree failed:" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)